app/javascript/packs/commentDropdowns.js

```diff
@@ -15,51 +15,99 @@ const handleCopyPermalink = (closeDropdown) => {
 };
 
 const initializeArticlePageDropdowns = () => {
+  // Gather all dropdown triggers for comment options and profile previews
   const dropdownTriggers = document.querySelectorAll(
-    'button[id^=comment-dropdown-trigger]',
+    'button[id^=comment-dropdown-trigger], button[id^=comment-profile-preview-trigger-]',
   );
+
   for (const dropdownTrigger of dropdownTriggers) {
     if (dropdownTrigger.dataset.initialized) {
       //  Make sure we only initialize once
       continue;
     }
 
     const dropdownContentId = dropdownTrigger.getAttribute('aria-controls');
+    const dropdownElement = document.getElementById(dropdownContentId);
 
-    const { closeDropdown } = initializeDropdown({
-      triggerElementId: dropdownTrigger.id,
-      dropdownContentId,
-    });
+    if (dropdownElement) {
+      const { closeDropdown } = initializeDropdown({
+        triggerElementId: dropdownTrigger.id,
+        dropdownContentId,
+      });
 
-    // Add actual link location (SEO doesn't like these "useless" links, so adding in here instead of in HTML)
-    const dropdownElement = document.getElementById(dropdownContentId);
-    const reportAbuseWrapper = dropdownElement.querySelector(
-      '.report-abuse-link-wrapper',
-    );
-    if (reportAbuseWrapper) {
-      reportAbuseWrapper.innerHTML = `<a href="${reportAbuseWrapper.dataset.path}" class="crayons-link crayons-link--block">Report abuse</a>`;
-    }
+      // Add actual link location (SEO doesn't like these "useless" links, so adding in here instead of in HTML)
+      const reportAbuseWrapper = dropdownElement.querySelector(
+        '.report-abuse-link-wrapper',
+      );
+      if (reportAbuseWrapper) {
+        reportAbuseWrapper.innerHTML = `<a href="${reportAbuseWrapper.dataset.path}" class="crayons-link crayons-link--block">Report abuse</a>`;
+      }
 
-    // Initialize the "Copy link" functionality
-    dropdownElement
-      .querySelector('.permalink-copybtn')
-      ?.addEventListener('click', handleCopyPermalink(closeDropdown));
+      // Initialize the "Copy link" functionality
+      dropdownElement
+        .querySelector('.permalink-copybtn')
+        ?.addEventListener('click', handleCopyPermalink(closeDropdown));
 
-    dropdownTrigger.dataset.initialized = 'true';
+      dropdownTrigger.dataset.initialized = 'true';
```

Comment:
Nitpick (non-blocking): The value can be set to `true` instead of `'true'`. It will become a string because of the nature of data-attributes.